test_runner/performance/test_physical_replication.py

```diff
@@ -0,0 +1,232 @@
+import subprocess
+import time
+from pathlib import Path
+from typing import Any, List, Optional
+
+import psycopg2
+import psycopg2.extras
+import pytest
+from fixtures.common_types import Lsn
+from fixtures.log_helper import log
+from fixtures.neon_fixtures import PgBin
+from fixtures.pg_version import PgVersion
+
+from performance.neon_api import (
+    neon_create_endpoint,
+    neon_create_project,
+    neon_delete_project,
+    neon_get_connection_uri,
+    neon_suspend_endpoint,
+    neon_wait_for_operation_to_finish,
+)
+
+
+# Granularity of ~0.5 sec
+def measure_replication_lag(master, replica):
+    start = time.time()
+    master.execute("SELECT pg_current_wal_flush_lsn()")
+    master_lsn = Lsn(master.fetchall()[0][0])
+    while True:
+        replica.execute("select pg_last_wal_replay_lsn()")
+        replica_lsn = replica.fetchall()[0][0]
+        if replica_lsn:
+            if Lsn(replica_lsn) >= master_lsn:
+                return time.time() - start
+        time.sleep(0.5)
+
+
+@pytest.mark.remote_cluster
+@pytest.mark.timeout(0)
```

Comment:
Let's put some timeout here

Reply:
What would be a good timeout? This is intended to be a several-hours long test

Reply:
Let's put several hours (whatever the default runtime is, plus some margin). On the CI we can override it if required

Reply:
Sounds good, I made tests run for 1 hour and gave timeout of 2 hours